lib/optimizer/tests/Transformer/AmpBoilerplateTest.php

```diff
@@ -57,65 +46,76 @@ public function dataTransform()
         $amp4EmailBoilerplate = '<style ' . Attribute::AMP4EMAIL_BOILERPLATE . '>' . Amp::AMP4ADS_AND_AMP4EMAIL_BOILERPLATE_CSS . '</style>';
 
         return [
-            'keeps boilerplate' => [
-                $inputWithBoilerplate('<html ⚡>', $ampBoilerplate),
-                $expected('<html ⚡>', $ampBoilerplate),
-            ],
+            'keeps boilerplate' => $repeatTwice(
+                $htmlDocument('<html ⚡>', $ampBoilerplate),
+            ),
 
-            'keeps boilerplate for amp4ads' => [
-                $inputWithBoilerplate('<html amp4ads>', $amp4AdsBoilerplate),
-                $expected('<html amp4ads>', $amp4AdsBoilerplate),
-            ],
+            'keeps boilerplate again' => $repeatTwice(
+                $htmlDocument('<html amp>', $ampBoilerplate),
+            ),
 
-            'keeps boilerplate for ⚡4ads' => [
-                $inputWithBoilerplate('<html ⚡4ads>', $amp4AdsBoilerplate),
-                $expected('<html ⚡4ads>', $amp4AdsBoilerplate),
-            ],
+            'leaves out boilerplate' => $repeatTwice(
+                $htmlDocument('<html amp i-amphtml-no-boilerplate>'),
```

Comment:
I noticed there weren't tests for the `i-amphtml-no-boilerplate` condition, so I've added a few here as well.